app/src/org/commcare/tasks/DataPullTask.java

```diff
@@ -217,7 +219,7 @@ public void reportProgress(int progress) {
                         CommCareApplication._().releaseUserResourcesAndServices();
                     }
                     Logger.log(AndroidLogger.TYPE_USER, "Bad Auth Request for user!|" + username);
-                    return PullTaskResult.AUTH_FAILED;
+                    return new Pair<>(PullTaskResult.AUTH_FAILED, "");

```

Comment:
Possibly too-cumbersome suggestion: instead of using a Pair with sometimes empty string, would it make sense to turn PullTaskResult from an enum into a full-fledged object with a String field and boolean hasMessage field?